### PR TITLE
add support for docker registry token auth server

### LINKED ISSUE #2316: Integrate or Interface Docker Registry
I know, this may be completely out of scope for this project. But I still try:

It would be cool if gitea could provide support for a docker registry.

Full integration is probably very unlikely to ever being added. But it would already help if we could interface with a standalone registry running in some container. The core features should be:

 * Authentication against shared user base
 * Authorization with shared repository permissions
 * Simple instructions for how to set up the above

### PR DESCRIPTION
**Pull request by dragonflylee**
Fix https://github.com/go-gitea/gitea/issues/2316 

similar to https://github.com/go-gitea/gitea/pull/14919 but only token server

* do not store registry repo to database
* fix auth scope type `registry:catalog:*`
* fix Unauthorized response message

ref:

* https://docs.docker.com/registry/spec/auth/token/
* https://github.com/distribution/distribution

### CODE REVIEW

**Comment by KN4CK3R** [comment]
Is it possible to use the existing JWT functionality? This would prevent another dependency:
https://github.com/go-gitea/gitea/blob/main/models/oauth2_application.go

File: options/locale/locale_zh-CN.ini [hunk 1]
```diff
@@ -1983,6 +1983,10 @@ topic.done=保存
 topic.count_prompt=您最多选择25个主题
 topic.format_prompt=主题必须以字母或数字开头，可以包含连字符 (-)，并且长度不得超过35个字符
 
+container = 容器仓库
```
Comment (lunny) [inline options/locale/locale_zh-CN.ini]: Don't change translation files except en-US. We use crowdin for translations.

**Comment by dragonflylee** [comment]
> Is it possible to use the existing JWT functionality? This would prevent another dependency:
> https://github.com/go-gitea/gitea/blob/main/models/oauth2_application.go

Yes it would be

File: options/locale/locale_zh-CN.ini (hunk 1 above)
Comment (dragonflylee) [reply options/locale/locale_zh-CN.ini]: remove in new commit

**Comment by lunny** [comment]
@dragonflylee It's better there is a repository level option like issue when global container is enabled.

**Comment by KN4CK3R** [comment]
You may not need a different private key, see #16010

**Review by lafriks — CHANGES_REQUESTED** [review]
We should reuse existing JWT functionality by adding needed scopes etc

**Comment by dragonflylee** [comment]
> You may not need a different private key, see #16010

registry only support rsa/ecdsa sign 
not all jwt auth method

**Comment by wxiaoguang** [comment]
This PR seems broken, many unrelated changes.

Either close this one and start a new clean one, or reset a clean branch from main and force push.

**Pull request closed without merging**